eth_abi/utils/numeric.py

```diff
@@ -51,3 +54,24 @@ def compute_signed_fixed_bounds(
         upper = decimal.Decimal(int_upper) * exp
 
     return lower, upper
+
+
+def scale_places(places: int) -> Callable[[decimal.Decimal], decimal.Decimal]:
+    """
+    Returns a function that shifts the decimal point of decimal values to the
+    right by ``places`` places.
+    """
+    with decimal.localcontext(abi_decimal_context):
+        scaling_factor = TEN ** int(-places)
```

Comment:
Maybe instead of casting, this should validate `places` and raise an exception if it's not already an `int`. (Or is there some legitimate case to use `places=0.5`?)

Reply:
Yeah, that's a better approach.